Sources/_CryptoExtras/RSA/RSA+BlindSigning.swift

```diff
@@ -0,0 +1,414 @@
+    public func prepare<D: DataProtocol>(_ message: D) -> _RSA.BlindSigning.PreparedMessage {
+        switch self.parameters.preparation {
+        case .identity:
+            return _RSA.BlindSigning.PreparedMessage(rawRepresentation: Data(message))
+        case .randomized:
+            return _RSA.BlindSigning.PreparedMessage(rawRepresentation: Data(SystemRandomNumberGenerator.randomBytes(count: 32)) + message)
+        }
+    }
+
+    /// Blind a message to be signed by the server using the blind signing protocol.
+    ///
+    /// - Parameter message: The message to be signed.
+    /// - Returns: The blinded message, and its inverse for unblinding its blind signature.
+    ///
+    /// - Seealso: [RFC 9474: Blind](https://www.rfc-editor.org/rfc/rfc9474.html#name-blind).
+    public func blind(
+        _ message: _RSA.BlindSigning.PreparedMessage
+    ) throws -> (blindedMessage: Data, blindInverse: _RSA.BlindSigning.BlindInverse) {
+        try self.backing.blind(message, parameters: self.parameters)
+    }
+
+    /// Unblinds the message and produce a signature for the message.
+    ///
+    /// - Parameter signature: The signature of the blinded message.
+    /// - Parameter message: The message to be signed.
+    /// - Parameter blindInverse: The inverse from the message blinding.
+    /// - Returns: The signature of the message.
+    ///
+    /// - Seealso: [RFC 9474: Finalize](https://www.rfc-editor.org/rfc/rfc9474.html#name-finalize).
+    public func finalize(
+        _ signature: _RSA.BlindSigning.BlindSignature,
+        for message: _RSA.BlindSigning.PreparedMessage,
+        blindInverse: _RSA.BlindSigning.BlindInverse
+    ) throws -> _RSA.Signing.RSASignature {
+        try self.backing.finalize(signature, for: message, blindInverse: blindInverse, parameters: self.parameters)
+    }
+
+    /// Validate a signature for a prepared message.
+    ///
+    /// - Parameter signature: The signature to verify.
+    /// - Parameter message: The prepared message used in the blind signature protocol.
+    /// - Returns: True if the signature is valid; false otherwise.
+    ///
+    /// - Seealso: [RFC 9474: Verification](https://www.rfc-editor.org/rfc/rfc9474.html#name-verification).
+    public func isValidSignature(
+        _ signature: _RSA.Signing.RSASignature,
+        for message: _RSA.BlindSigning.PreparedMessage
+    ) -> Bool {
+        switch parameters.padding {
+        case .PSS:
+            return self.backing.isValidSignature(signature, for: H.hash(data: message.rawRepresentation), padding: .PSS)
+        case .PSSZERO:
+            return self.backing.isValidSignature(signature, for: H.hash(data: message.rawRepresentation), padding: .PSSZERO)
+        }
+    }
+}
+
+extension _RSA.BlindSigning {
+    /// Errors defined in the RSA Blind Signatures protocol.
+    ///
+    /// - Seealso: [RFC 9474: Errors](https://www.rfc-editor.org/rfc/rfc9474.html#name-errors).
+    enum Error: Swift.Error {
```

Comment:
This error should probably be public, and probably not be an enum.

Reply:
I have a local patch for this suggestion but I haven't pushed it yet because I wanted to discuss it more. What do we expect people to do with these errors? Just be able to branch on the different ones? For that, I guess we need an enum-like struct wrapper with some public lets?

Reply:
OK, I've made sure that we only throw the existing, public `CryptoKitError` through the API.